src/components/AccountForm.js

```diff
@@ -0,0 +1,103 @@
+import React, {useState} from 'react';
+import {Link, useHistory} from 'react-router-dom';
+
+const AccountForm = ({type, setToken, setUser}) => {
+    const [username, setUsername] = useState('');
```

Comment:
No required, but think about putting state together in an object if it's related... like `user` data...

```js
const [userFormData, setUserFormData] = useState({
username: '',
...
});
// ...
setUserFormData({...userFormData, password});
```